### ISSUE TITLE
Exit status of command 'swiftlint lint --config .swiftlint.yml' was 1 instead of 0.

### ISSUE THREAD

**Issue by odemolliens** [fastlane/fastlane#1442]
### New Issue Checklist

- [V] Updated fastlane to the latest version
- [V] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

##### Complete output when running fastlane, including the stack trace and command used

```
from /usr/local/lib/ruby/gems/2.2.0/gems/fastlane_core-0.36.8/lib/fastlane_core/ui/ui.rb:14:in `method_missing'
	from /usr/local/lib/ruby/gems/2.2.0/gems/fastlane-1.63.1/lib/fastlane/helper/sh_helper.rb:44:in `sh_no_action'
	from /usr/local/lib/ruby/gems/2.2.0/gems/fastlane-1.63.1/lib/fastlane/helper/sh_helper.rb:9:in `sh'
	from /usr/local/lib/ruby/gems/2.2.0/gems/fastlane-1.63.1/lib/fastlane/actions/swiftlint.rb:30:in `run'
	from /usr/local/lib/ruby/gems/2.2.0/gems/fastlane-1.63.1/lib/fastlane/runner.rb:143:in `block (2 levels) in execute_action'
	from /usr/local/lib/ruby/gems/2.2.0/gems/fastlane-1.63.1/lib/fastlane/actions/actions_helper.rb:35:in `execute_action'
	from /usr/local/lib/ruby/gems/2.2.0/gems/fastlane-1.63.1/lib/fastlane/runner.rb:129:in `block in execute_action'
	from /usr/local/lib/ruby/gems/2.2.0/gems/fastlane-1.63.1/lib/fastlane/runner.rb:128:in `chdir'
	from /usr/local/lib/ruby/gems/2.2.0/gems/fastlane-1.63.1/lib/fastlane/runner.rb:128:in `execute_action'
	from /usr/local/lib/ruby/gems/2.2.0/gems/fastlane-1.63.1/lib/fastlane/fast_file.rb:149:in `method_missing'
	from Fastfile:288:in `block in parsing_binding'
	from /usr/local/lib/ruby/gems/2.2.0/gems/fastlane-1.63.1/lib/fastlane/lane.rb:36:in `call'
	from /usr/local/lib/ruby/gems/2.2.0/gems/fastlane-1.63.1/lib/fastlane/lane.rb:36:in `call'
	from /usr/local/lib/ruby/gems/2.2.0/gems/fastlane-1.63.1/lib/fastlane/runner.rb:52:in `block in execute'
	from /usr/local/lib/ruby/gems/2.2.0/gems/fastlane-1.63.1/lib/fastlane/runner.rb:46:in `chdir'
	from /usr/local/lib/ruby/gems/2.2.0/gems/fastlane-1.63.1/lib/fastlane/runner.rb:46:in `execute'
	from /usr/local/lib/ruby/gems/2.2.0/gems/fastlane-1.63.1/lib/fastlane/lane_manager.rb:46:in `cruise_lane'
	from /usr/local/lib/ruby/gems/2.2.0/gems/fastlane-1.63.1/lib/fastlane/command_line_handler.rb:30:in `handle'
	from /usr/local/lib/ruby/gems/2.2.0/gems/fastlane-1.63.1/bin/fastlane:38:in `block (2 levels) in run'
	from /usr/local/lib/ruby/gems/2.2.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /usr/local/lib/ruby/gems/2.2.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /usr/local/lib/ruby/gems/2.2.0/gems/commander-4.3.5/lib/commander/command.rb:153:in `run'
	from /usr/local/lib/ruby/gems/2.2.0/gems/commander-4.3.5/lib/commander/runner.rb:428:in `run_active_command'
	from /usr/local/lib/ruby/gems/2.2.0/gems/fastlane_core-0.36.8/lib/fastlane_core/ui/fastlane_runner.rb:23:in `run!'
	from /usr/local/lib/ruby/gems/2.2.0/gems/commander-4.3.5/lib/commander/delegates.rb:15:in `run!'
	from /usr/local/lib/ruby/gems/2.2.0/gems/fastlane-1.63.1/bin/fastlane:156:in `run'
	from /usr/local/lib/ruby/gems/2.2.0/gems/fastlane-1.63.1/bin/fastlane:162:in `<top (required)>'
	from /usr/local/bin/fastlane:23:in `load'
	from /usr/local/bin/fastlane:23:in `<main>'
```

### Configuration Files

Please copy the complete content of your `Fastfile` and any other configuration files you use below:

Fastfile:
```ruby
  swiftlint(
    mode: :lint,
    strict: false,
    config_file: '.swiftlint.yml',
  )
```
AND/OR

```ruby
  swiftlint(
    mode: :lint,
    strict: false,
  )
```

### Environment

fastlane version (fastlane -v): fastlane **1.63.1**
Do you use bundler, rbenv or rvm? **NO**

I'm trying to find a way to generate **swiftlint report file** without **broke** the ci (jenkins) build

Actually I get this result: `Valid Done linting! Found 261 violations, 35 serious in 28 files. `

and fastlane throw:
```
/usr/local/lib/ruby/gems/2.2.0/gems/fastlane_core-0.36.8/lib/fastlane_core/ui/interface.rb:120:in `crash!': [!] Exit status of command 'swiftlint lint --config .swiftlint.yml' was 1 instead of 0. (FastlaneCore::Interface::FastlaneCrash)

```
Without config file (--config .swiftlint.yml):

```
/usr/local/lib/ruby/gems/2.2.0/gems/fastlane_core-0.36.8/lib/fastlane_core/ui/interface.rb:120:in `crash!': [!] Exit status of command 'swiftlint lint' was 1 instead of 0. (FastlaneCore::Interface::FastlaneCrash)
```

**Comment by vpolouchkine** [comment]
Does it work outside of `fastlane`?

**Comment by odemolliens** [comment]
yes without any problems

**Comment by odemolliens** [comment]
Up

**Comment by odemolliens** [comment]
Up

**Comment by vpolouchkine** [comment]
Perhaps @HeEAaD might know?

**Comment by KrauseFx** [comment]
```
/usr/local/lib/ruby/gems/2.2.0/gems/fastlane_core-0.36.8/lib/fastlane_core/ui/interface.rb:120:in `crash!': [!] Exit status of command 'swiftlint lint --config .swiftlint.yml' was 1 instead of 0. (FastlaneCore::Interface::FastlaneCrash)
```

This is perfectly valid: the `swiftlint` command returns an exit status code that indicates and error and therefore `fastlane` will show that it failed. This is not an issue with `fastlane`

**Comment by palusik86** [comment]
This is issue of fastlane, when I run command `swiftlint lint --config fastlane/rules.yml` it works perfectly, but when I used localy swiflint i got:
` Done linting! Found 4268 violations, 475 serious in 414 files`.

and just after that 

`Exit status of command 'swiftlint lint --config fastlane/rules.yml' was 1 instead of 0.`

and dump from swiftling logs, all files etc. 

Last lines:

```
Done linting! Found 4268 violations, 475 serious in 414 files.
()

	from /Library/Ruby/Gems/2.0.0/gems/fastlane_core-0.39.0/lib/fastlane_core/ui/ui.rb:15:in `method_missing'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.66.0/lib/fastlane/helper/sh_helper.rb:44:in `sh_no_action'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.66.0/lib/fastlane/helper/sh_helper.rb:9:in `sh'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.66.0/lib/fastlane/actions/swiftlint.rb:30:in `run'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.66.0/lib/fastlane/runner.rb:143:in `block (2 levels) in execute_action'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.66.0/lib/fastlane/actions/actions_helper.rb:35:in `execute_action'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.66.0/lib/fastlane/runner.rb:129:in `block in execute_action'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.66.0/lib/fastlane/runner.rb:128:in `chdir'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.66.0/lib/fastlane/runner.rb:128:in `execute_action'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.66.0/lib/fastlane/fast_file.rb:149:in `method_missing'
	from GenericFastfile:180:in `block (2 levels) in parsing_binding'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.66.0/lib/fastlane/lane.rb:36:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.66.0/lib/fastlane/lane.rb:36:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.66.0/lib/fastlane/runner.rb:52:in `block in execute'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.66.0/lib/fastlane/runner.rb:46:in `chdir'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.66.0/lib/fastlane/runner.rb:46:in `execute'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.66.0/lib/fastlane/lane_manager.rb:46:in `cruise_lane'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.66.0/lib/fastlane/command_line_handler.rb:30:in `handle'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.66.0/bin/fastlane:38:in `block (2 levels) in run'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/command.rb:153:in `run'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/runner.rb:428:in `run_active_command'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane_core-0.39.0/lib/fastlane_core/ui/fastlane_runner.rb:23:in `run!'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/delegates.rb:15:in `run!'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.66.0/bin/fastlane:156:in `run'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.66.0/bin/fastlane:162:in `<top (required)>'
	from /usr/local/bin/fastlane:23:in `load'
	from /usr/local/bin/fastlane:23:in `<main>'
```

**Comment by odemolliens** [comment]
@palusik86 Yeah I totally agree with you. That's why I have open this ticket

**Comment by KrauseFx** [comment]
Maybe you're running it in the wrong directory? Try putting this around `swiftline`
```ruby
Dir.chdir("..") do
  swiftlint
end
```

**Comment by odemolliens** [comment]
Linting is working as well as you can see on previous page

Maybe I can add some debug logs on swiflint.rb to help to find the bug?

**Comment by palusik86** [comment]
I will do more tests run today and post some more info.

**Comment by mgasiorowski** [comment]
Swiftlint always returns `exit 1` when he founds serious violations (https://github.com/realm/SwiftLint/issues/30), but I think it should not fail job on CI. For example Jenkins has plugins for inform about violations (e.g. https://wiki.jenkins-ci.org/display/JENKINS/Checkstyle+Plugin) and swiftlint support custom reporters,
https://github.com/realm/SwiftLint/pull/293/files.

**Comment by palusik86** [comment]
Will try, Thanks

**Comment by mfurtak** [comment]
Hi folks. I've merged #4112, which adds a `ignore_exit_status` option to the [`swiftlint` action](https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md#swiftlint) that should do what you're looking for.

That will go out in the next release! :rocket:

**Comment by mfurtak** [comment]
This has gone out as part of [`fastlane` 1.81.0](https://github.com/fastlane/fastlane/releases/tag/1.81.0)! :tada:
